wp-includes/js/service-worker-navigation-routing.js

```diff
@@ -1,121 +1,140 @@
 
-const isStreamingResponses = SHOULD_STREAM_RESPONSE && wp.serviceWorker.streams.isSupported();
+	wp.serviceWorker.routing.registerRoute( new wp.serviceWorker.routing.NavigationRoute(
+		async function ( { event } ) {
+			const { url } = event.request;
 
-wp.serviceWorker.routing.registerRoute( new wp.serviceWorker.routing.NavigationRoute(
-	async function ( { event } ) {
-		const { url } = event.request;
+			let responsePreloaded = false;
 
-		let responsePreloaded = false;
-
-		const canStreamResponse = () => {
-			return isStreamingResponses && ! responsePreloaded;
-		};
+			const canStreamResponse = () => {
+				return isStreamingResponses && ! responsePreloaded;
+			};
 
-		const handleResponse = ( response ) => {
-			if ( response.status < 500 ) {
-				if ( response.redirected ) {
-					const redirectedUrl = new URL( response.url );
-					redirectedUrl.searchParams.delete( STREAM_HEADER_FRAGMENT_QUERY_VAR );
-					const script = `
+			const handleResponse = ( response ) => {
+				if ( response.status < 500 ) {
+					if ( response.redirected ) {
+						const redirectedUrl = new URL( response.url );
+						redirectedUrl.searchParams.delete( STREAM_HEADER_FRAGMENT_QUERY_VAR );
+						const script = `
 						<script id="wp-stream-fragment-replace-state">
 						history.replaceState( {}, '', ${ JSON.stringify( redirectedUrl.toString() ) } );
 						document.getElementById( 'wp-stream-fragment-replace-state' ).remove();
 						</script>
 					`;
-					return response.text().then( ( body ) => {
-						return new Response( script + body );
-					} );
-				} else {
-					return response;
+						return response.text().then( ( body ) => {
+							return new Response( script + body );
+						} );
+					} else {
+						return response;
+					}
 				}
-			}
-			const channel = new BroadcastChannel( 'wordpress-server-errors' );
-
-			// Wait for client to request the error message.
-			channel.onmessage = ( event ) => {
-				if ( event.data && event.data.clientUrl && url === event.data.clientUrl ) {
-					response.text().then( ( text ) => {
-						channel.postMessage({
-							requestUrl: url,
-							bodyText: text,
+				const channel = new BroadcastChannel( 'wordpress-server-errors' );
+
+				// Wait for client to request the error message.
+				channel.onmessage = ( event ) => {
+					if ( event.data && event.data.clientUrl && url === event.data.clientUrl ) {
+						response.text().then( ( text ) => {
+							channel.postMessage({
+								requestUrl: url,
+								bodyText: text,
+								status: response.status,
+								statusText: response.statusText
+							});
+							channel.close();
+						} );
+					}
+				};
+
+				// Close the channel if client did not request the message within 30 seconds.
+				setTimeout( () => {
+					channel.close();
+				}, 30 * 1000 );
+
+				return caches.match( canStreamResponse() ? ERROR_500_BODY_FRAGMENT_URL : ERROR_500_URL );
+			};
+
+			const sendOfflineResponse = () => {
+				if ( canStreamResponse() ) {
```

Comment:
Here starts the actual main change for the file, most of the other changes are related to wrapping it into curly brackets.